platform/ios/app-swift/Sources/LocationPrivacyExample.swift

```diff
@@ -9,75 +9,114 @@ enum LocationAccuracyState {
     case fullAccuracy
 }
 
-class Coordinator: NSObject, MLNMapViewDelegate {
-    @Binding var mapView: MLNMapView
-    @Binding var locationAccuracy: LocationAccuracyState
-    var pannedToUserLocation = false
-
-    init(mapView: Binding<MLNMapView>, locationAccuracy: Binding<LocationAccuracyState>) {
-        _mapView = mapView
-        _locationAccuracy = locationAccuracy
+class MapViewModel: NSObject, ObservableObject {
+    // Weak because the coordinator has a strong reference to the view model
+    weak var mapCoordinater: MapLibreRepresentableCoordinator?
+    @MainActor @Published var locationAccuracy: LocationAccuracyState = .unknown
+
+    @MainActor func requestTemporaryLocationAuthorization() {
+        print("Requesting precice location")
+
+        switch locationAccuracy {
+        case .reducedAccuracy:
+            let purposeKey = "MLNAccuracyAuthorizationDescription"
+            mapCoordinater?.mapView?.locationManager.requestTemporaryFullAccuracyAuthorization!(withPurposeKey: purposeKey)
+        default:
+            break
+        }
     }
+}
 
-    func mapView(_: MLNMapView, didChangeLocationManagerAuthorization manager: MLNLocationManager) {
+class MapLibreRepresentableCoordinator: NSObject, MLNMapViewDelegate {
+    private var mapViewModel: MapViewModel
+    // Weak reference because SwiftUI owns the strong reference
+    private(set) weak var mapView: MLNMapView?
+    private var pannedToUserLocation = false
+
+    init(mapViewModel: MapViewModel) {
+        self.mapViewModel = mapViewModel
+        super.init()
+        self.mapViewModel.mapCoordinater = self
+    }
+
+    @MainActor func mapView(_: MLNMapView, didChangeLocationManagerAuthorization manager: MLNLocationManager) {
         guard let accuracySetting = manager.accuracyAuthorization else {
             return
         }
 
         switch accuracySetting() {
         case .fullAccuracy:
-            locationAccuracy = .fullAccuracy
+            mapViewModel.locationAccuracy = .fullAccuracy
         case .reducedAccuracy:
-            locationAccuracy = .reducedAccuracy
+            mapViewModel.locationAccuracy = .reducedAccuracy
         @unknown default:
-            locationAccuracy = .unknown
+            mapViewModel.locationAccuracy = .unknown
         }
     }
 
     // when a location is available for the first time, we fly to it
     func mapView(_ mapView: MLNMapView, didUpdate _: MLNUserLocation?) {
-        if pannedToUserLocation {
-            return
-        }
+        guard !pannedToUserLocation else { return }
         guard let userLocation = mapView.userLocation else {
             print("User location is currently not available.")
             return
         }
         mapView.fly(to: MLNMapCamera(lookingAtCenter: userLocation.coordinate, altitude: 100_000, pitch: 0, heading: 0))
+        pannedToUserLocation = true
+    }
+
+    func mapView(_ mapView: MLNMapView, didFinishLoading _: MLNStyle) {
```

Comment:
Are these two methods strictly required?

I'd like to keep the example as focused as possible.

Reply:
Nope, removing, done.